week7/answers.sql

```diff
@@ -1,3 +1,306 @@
+);
+
+CREATE TABLE inventory (
+  inventory_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  character_id INT UNSIGNED NOT NULL,
+  item_id INT UNSIGNED NOT NULL,
+  CONSTRAINT inventory_fk_characters
+  FOREIGN KEY (character_id) 
+    REFERENCES characters (character_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE,       
+    CONSTRAINT inventory_fk_items
+    FOREIGN KEY (item_id) 
+       REFERENCES items (item_id)
+       ON UPDATE CASCADE
+       ON DELETE CASCADE       
+);
+
+CREATE TABLE equipped (
+  equipped_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  character_id INT UNSIGNED NOT NULL,
+  item_id INT UNSIGNED NOT NULL,
+  CONSTRAINT equipped_fk_characters
+  FOREIGN KEY (character_id) 
+     REFERENCES characters (character_id)
+     ON UPDATE CASCADE
+     ON DELETE CASCADE,      
+   CONSTRAINT equipped_fk_items
+	FOREIGN KEY (item_id) 
+        REFERENCES items (item_id)
+         ON UPDATE CASCADE
+	 ON DELETE CASCADE
+);
+
+CREATE OR REPLACE VIEW character_items AS
+SELECT c.character_id, c.name AS character_name, i.name AS item_name, i.armor , i.damage
+FROM characters c
+    INNER JOIN inventory iv
+    ON c.character_id = iv.character_id
+    INNER JOIN items i
+    ON iv.item_id = i.item_id
+    UNION 
+    SELECT c.character_id, c.name AS character_name, i.name AS item_name, i.armor , i.damage
+    FROM characters c
+    INNER JOIN equipped e
+    ON c.character_id = e.character_id
+    INNER JOIN items i
+    ON e.item_id = i.item_id 
+    ORDER BY character_name, item_name;
+    
+CREATE OR REPLACE VIEW team_items AS
+    SELECT t.team_id, t.name AS team_name, i.name AS item_name, i.armor , i.damage
+    FROM teams t
+    INNER JOIN team_members te
+    ON t.team_id = te.team_id
+    INNER JOIN characters c
+    ON c.character_id = te.character_id
+    INNER JOIN inventory iv
+    ON c.character_id = iv.character_id
+    INNER JOIN items i
+    ON iv.item_id = i.item_id
+    UNION 
+    SELECT t.team_id, t.name AS team_name, i.name AS item_name, i.armor , i.damage
+    FROM teams t
+    INNER JOIN team_members te
+    ON t.team_id = te.team_id
+    INNER JOIN characters c
+    ON c.character_id = te.character_id
+    INNER JOIN equipped e
+    ON c.character_id = e.character_id
+    INNER JOIN items i
+    ON e.item_id = i.item_id 
+    ORDER BY team_name, item_name;
+    
+DELIMITER ;;
+    CREATE FUNCTION armor_total(id INT UNSIGNED)
+    RETURNS INT UNSIGNED 
+    DETERMINISTIC 
+    BEGIN 
+    DEClARE character_armor INT UNSIGNED;
+    DEClARE equipped_armor INT UNSIGNED;
+    DECLARE total_armor INT UNSIGNED;
+
+    SELECT armor  INTO character_armor
+    FROM character_stats 
+    WHERE character_id = id;
+
+    SELECT SUM(i.armor) INTO equipped_armor 
+    FROM items i
+    LEFT OUTER JOIN equipped e
+    ON e.item_id = i.item_id
+    WHERE e.character_id = id;
+
+    SET total_armor = character_armor + equipped_armor;
+        RETURN total_armor;
+
+    END;; 
+     
+DELIMITER ;;
```

Comment:
You can wrap all of your FUNCTIONs and PROCEDUREs into a single DELIMITER pair. For example:

```sql
DELIMITER ;;
CREATE FUNCTION armor_total ...
   ...
CREATE PROCEDURE attack ...
   ...
DELIMITER ;
```